samza-api/src/main/java/org/apache/samza/metrics/RegistryWithSource.java

```diff
@@ -16,28 +16,26 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+package org.apache.samza.metrics;
 
-package org.apache.samza.metrics.reporter
-
-import java.util.Map
-import java.util.HashMap
-import scala.beans.BeanProperty
+/**
+ * Simple holder for {@link ReadableMetricsRegistry} and its source, used for implementations of
+ * {@link MetricsReporter}.
+ */
+public class RegistryWithSource {
```

Comment:
I prefer to not use `Pair` because it isn't as explicit about what is inside of it. It is usually not too hard to figure out what `getLeft` and `getRight` returns, but it is sometimes not obvious, especially when one of the items is a primitive type like a `String`. In this case, if `Pair` was used, then a reader would need to trace back through to where the `Pair` was added to the list in order to figure out that the `String` part of the `Pair` is "source". Although it is an extra class, I prefer the explicit readability.

Reply:
Yes, the registry container is implemented as a list, so it is possible for the same source to be associated with two different registries.